packages/keyring-eth-hd/src/hd-keyring-v2.ts

```diff
@@ -0,0 +1,402 @@
+      throw new Error(
+        `Can only create the next account in sequence. ` +
+          `Expected groupIndex ${currentCount}, got ${targetIndex}.`,
+      );
+    }
+
+    // Add the next account
+    const newAddresses = await this.inner.addAccounts(1);
+    const newAddress = newAddresses[0];
+
+    if (!newAddress) {
+      throw new Error('Failed to create new account');
+    }
+
+    const newAccount = this.#createKeyringAccount(newAddress, targetIndex);
+
+    return [newAccount];
+  }
+
+  /**
+   * Delete an account from the keyring.
+   *
+   * ⚠️ Warning: Only deleting the last account is possible.
+   *
+   * @param accountId - The account ID to delete.
+   */
+  async deleteAccount(accountId: AccountId): Promise<void> {
+    // Get the account first, before any registry operations
+    const { address } = await this.getAccount(accountId);
+    const hexAddress = this.#toHexAddress(address);
+
+    // Assert that the account to delete is the last one in the inner keyring
+    // We check against the inner keyring directly to avoid stale registry issues
+    if (!(await this.#isLastAccount(address))) {
+      throw new Error(
+        'Can only delete the last account in the HD keyring due to derivation index constraints.',
+      );
+    }
+
+    // Remove from the legacy keyring
+    this.inner.removeAccount(hexAddress);
+
+    // Remove from the registry
+    this.registry.delete(accountId);
+  }
+
+  async exportAccount(
+    accountId: AccountId,
+    options?: ExportAccountOptions,
+  ): Promise<ExportedAccount> {
+    const account = await this.getAccount(accountId);
+
+    // Validate encoding - we only support hexadecimal for Ethereum keys
+    const requestedEncoding =
+      options?.encoding ?? PrivateKeyEncoding.Hexadecimal;
+
+    if (requestedEncoding !== PrivateKeyEncoding.Hexadecimal) {
+      throw new Error(
+        `Unsupported encoding for Ethereum HD keyring: ${requestedEncoding}. Only '${PrivateKeyEncoding.Hexadecimal}' is supported.`,
+      );
+    }
+
+    // The legacy HdKeyring returns a hex string without 0x prefix.
+    const privateKeyHex = await this.inner.exportAccount(
+      this.#toHexAddress(account.address),
+    );
+
+    const exported: ExportedAccount = {
+      type: 'private-key',
+      privateKey: `0x${privateKeyHex}`,
+      encoding: PrivateKeyEncoding.Hexadecimal,
+    };
+
+    return exported;
+  }
+
+  async submitRequest(request: KeyringRequest): Promise<Json> {
+    const { method, params = [] } = request.request;
+
+    const { address, methods } = await this.getAccount(request.account);
+    const hexAddress = this.#toHexAddress(address);
+
+    // Validate account can handle the method
+    if (!methods.includes(method)) {
+      throw new Error(
+        `Account ${request.account} cannot handle method: ${method}`,
+      );
+    }
+
+    // Validate params is an array
+    if (!Array.isArray(params)) {
+      throw new Error('Expected params to be an array');
+    }
+
```

Comment:
Not sure we need this actually, the `assert` below should already take care of that, and they should give a pretty good error details for it too!

```suggestion
```